### PR TITLE
docs: reduce cumulative layout shift

### PR DESCRIPTION
**Pull request by amareshsm**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:
     Reduced cumulative layout shift.

#### What changes did you make? (Give an overview)
 Reduced cumulative layout shift.
     

|   | Before | After  |
| ------------- | ------------- | -------------|
| Desktop | Screenshot 2022-06-24 at 1 12 42 AM | Screenshot 2022-06-24 at 1 06 57 AM |
|  Mobile | Screenshot 2022-06-24 at 1 09 35 AM | Screenshot 2022-06-24 at 1 07 38 AM |

### CODE REVIEW

**Review by snitin315 — APPROVED** [review]
LGTM. Thanks!

File: docs/src/assets/scss/carbon-ads.scss [hunk 1]
```diff
@@ -4,6 +4,10 @@
      }
 }
 
+.docs-ad {
+    height: 335px;
+}
```
Comment (mdjermanovic) [inline docs/src/assets/scss/carbon-ads.scss:9]: I understand why we want to set `height` here, but how did we calculate that it should be `335px`?

Comment (mdjermanovic) [reply docs/src/assets/scss/carbon-ads.scss:9]: On the eslint.org homepage, it's set to [290px](https://github.com/eslint/eslint.org/blob/1f3d6e58af1226b3aa62d10cdf9667a608d18c92/src/assets/scss/components/carbon-ads.scss#L13)?

Comment (mdjermanovic) [reply docs/src/assets/scss/carbon-ads.scss:9]: @nzakas do we know (or have control over it) what the maximum ad height is?

Comment (nzakas) [reply docs/src/assets/scss/carbon-ads.scss:9]: I’m pretty sure the max height is 290px. I remember digging into this when we were in the middle of the redesign. We definitely don’t want different settings between the docs site and the main site. I’ll take a closer look at this next week.

**Comment by nzakas** [comment]
@SaraSoueidan thoughts on this?

File: docs/src/assets/scss/components/docs-index.scss [hunk 2]
```diff
@@ -152,3 +157,33 @@
...
+.no-js #js-docs-index-panel {
+    @media all and (max-width: 1023px) {
+        display: block !important;
+    } 
+}
+
+.no-js #js-toc-panel {
+    @media all and (max-width: 1023px) {
+        display: block !important;
+    }
```
Comment (harish-sethuraman) [inline docs/src/assets/scss/components/docs-index.scss:188]: small nit:
```suggestion
@media all and (max-width: 1023px) {
    .no-js #js-docs-index-panel,
    .no-js #nav-panel,
    .no-js #js-toc-panel {
        display: block !important;
    }
}
```

File: docs/src/assets/scss/components/docs-navigation.scss [hunk 3]
```diff
@@ -71,8 +78,32 @@
...
+    @media all and (max-width: 1023px) {
+        display: none;
+    }
+}
+
+#js-docs-index-list[data-open="true"]{
+    @media all and (max-width: 1023px) {
+    display: block;
+    }
+}
```
Comment (harish-sethuraman) [inline docs/src/assets/scss/components/docs-navigation.scss:103]: small nit: We could have extracted the media query outside and could have had all the selectors + attributes + values inside the same media query?  I see that all media queries are common?

File: docs/src/assets/scss/components/docs-index.scss [hunk 4]
```diff
@@ -152,3 +157,33 @@
         }
     }
 }
+
+.enhanced .switcher-fallback {
+    display: none !important
```
Comment (nzakas) [inline docs/src/assets/scss/components/docs-index.scss:162]: I’m not a fan of needing to rely on !important everywhere.

Could we simplify all of this logic by maybe creating .js-only and .js-only-medium classes that we can just apply in the HTML?

**Comment by nzakas** [comment]
Closing due to age.

**Pull request closed without merging**